cpp/src/gandiva/precompiled/decimal_wrapper.cc

```diff
@@ -229,14 +229,43 @@ void castDECIMAL_float32(float in, int32_t x_precision, int32_t x_scale,
 }
 
 FORCE_INLINE
-void castDECIMAL_decimal128(int64_t x_high, uint64_t x_low, int32_t x_precision,
-                            int32_t x_scale, int32_t out_precision, int32_t out_scale,
-                            int64_t* out_high, int64_t* out_low) {
+boolean castDecimal_return_overflow(int64_t x_high, uint64_t x_low, int32_t x_precision,
```

Comment:
can you please name it as castDecimal_internal ? having the return value in name looks odd.

Reply:
done